starknet/test/stark-sig-auth.test.ts

```diff
@@ -1,256 +1,499 @@
+      domain: domain,
+      message: voteMsg as any,
+    } as typedData.TypedData)) as any;
+
+    const voteCalldata = CallData.compile({
+      signature: [voteSig.r, voteSig.s],
+      ...voteMsg,
+      accountType: accountType,
+    });
+
+    await account.invoke(starkSigAuthenticator, 'authenticate_vote', voteCalldata, {
+      rawInput: true,
     });
+  }, 1000000);
 
-    {
-      // Random Tx just to advance the block number on the devnet so the voting period begins.
-
-      await account0.declareAndDeploy({
-        contract: json.parse(
-          fs
-            .readFileSync('starknet/target/dev/sx_StarkSigAuthenticator.sierra.json')
-            .toString('ascii'),
-        ),
-        casm: json.parse(
-          fs
-            .readFileSync('starknet/target/dev/sx_StarkSigAuthenticator.casm.json')
-            .toString('ascii'),
-        ),
-        constructorCalldata: CallData.compile({ name: 'sx-sn', version: '0.1.0' }),
-      });
+  it('should revert if an incorrect signature is used', async () => {
+    await starknet.devnet.restart();
+    await starknet.devnet.load('./dump.pkl');
+    await starknet.devnet.increaseTime(10);
+
+    // Account #1 on Starknet devnet with seed 42
+    const invalidAccountWithSigner = new Account(
+      new Provider({ sequencer: { baseUrl: network } }),
+      '0x7aac39162d91acf2c4f0d539f4b81e23832619ac0c3df9fce22e4a8d505632a',
+      '0x23b8c1e9392456de3eb13b9046685257',
+    );
+
+    // PROPOSE
+    const proposeMsg: Propose = {
+      space: space.address,
+      author: account.address,
+      metadataUri: ['0x1', '0x2', '0x3', '0x4'],
+      executionStrategy: {
+        address: '0x0000000000000000000000000000000000005678',
+        params: ['0x0'],
+      },
+      userProposalValidationParams: [
+        '0xffffffffffffffffffffffffffffffffffffffffff',
+        '0x1234',
+        '0x5678',
+        '0x9abc',
+      ],
+      salt: '0x0',
+    };
+
+    const invalidProposeSig = (await invalidAccountWithSigner.signMessage({
+      types: proposeTypes,
+      primaryType: 'Propose',
+      domain: domain,
+      message: proposeMsg as any,
+    } as typedData.TypedData)) as any;
+
+    const invalidProposeCalldata = CallData.compile({
+      signature: [invalidProposeSig.r, invalidProposeSig.s],
+      ...proposeMsg,
+      accountType: accountType,
+    });
 
-      await account0.declareAndDeploy({
-        contract: json.parse(
-          fs
-            .readFileSync('starknet/target/dev/sx_StarkSigAuthenticator.sierra.json')
-            .toString('ascii'),
-        ),
-        casm: json.parse(
-          fs
-            .readFileSync('starknet/target/dev/sx_StarkSigAuthenticator.casm.json')
-            .toString('ascii'),
-        ),
-        constructorCalldata: CallData.compile({ name: 'sx-sn', version: '0.1.0' }),
+    try {
+      await account.invoke(starkSigAuthenticator, 'authenticate_propose', invalidProposeCalldata, {
+        rawInput: true,
```

Comment:
should throw an error after the `await` so that if it actually succeeds, the test still fails. Same for the calls below

Reply:
ah yeah adding - also for eth sig test.